### PR TITLE
Cryp 187/change mercat dependencies

### PR DESCRIPTION
**Pull request by aafshar-poly**
Changes to dependencies to match the restructure of cryptography repo done in https://github.com/PolymathNetwork/cryptography/pull/97

I'll fix the CI issues and then point it to the correct branch

### CODE REVIEW

File: pallets/confidential/Cargo.toml [hunk 1]
```diff
@@ -30,30 +30,20 @@ sp-runtime-interface = { git = "https://github.com/paritytech/substrate", defaul
 # Crypto
 rand_core = { version = "0.5", default-features = false }
 rand = { version = "0.7.3", default-features = false, optional = true }
-cryptography = { git = "https://github.com/PolymathNetwork/cryptography.git", branch = "develop", default-features = false }
-curve25519-dalek = { git = "https://github.com/PolymathNetwork/curve25519-dalek.git", branch = "v2-packed-simd", default-features = false, features = ["nightly"] }
-bulletproofs = { git = "https://github.com/PolymathNetwork/bulletproofs.git", branch = "v2-packed-simd", default-features = false, features = ["zeroize"] }
+cryptography_core = { git = "https://github.com/PolymathNetwork/cryptography.git", branch = "CRYP-187/cleanup-dir-structure_miguel", default-features = false }
```
Comment (ParnianAlimi) [inline pallets/confidential/Cargo.toml:33]: I can't wait for this `confidential` pallet to be removed. It's essentially accessing the bulletproofs library via the cryptography library... And it's dead code...

Comment (fmiguelgarcia) [reply pallets/confidential/Cargo.toml:33]: Actually @ParnianAlimi you could delete it. This pallet was created as a E2E test for the integration with crypto-library.
In order to delete it, pls be aware that you should migrate its benchmarks too.

Comment (ParnianAlimi) [reply pallets/confidential/Cargo.toml:33]: Yeah, I brought this up a while ago. and Adam said it's for the future.

File: pallets/runtime/tests/src/settlement_test.rs [hunk 2]
```diff
@@ -11,23 +11,22 @@ use confidential_asset::{
...
-    mercat::{
-        account::{convert_asset_ids, AccountCreator},
-        asset::AssetIssuer,
-        transaction::{CtxMediator, CtxReceiver, CtxSender},
-        Account, AccountCreatorInitializer, AssetTransactionIssuer, EncryptedAmount,
-        EncryptionKeys, FinalizedTransferTx, InitializedTransferTx, PubAccount, PubAccountTx,
-        SecAccount, TransferTransactionMediator, TransferTransactionReceiver,
-        TransferTransactionSender,
-    },
+    curve25519_dalek::scalar::Scalar,
```
Comment (ParnianAlimi) [inline pallets/runtime/tests/src/settlement_test.rs:16]: I think we should access these via the `mercat` dependency at line 22 now. In order to bundle `cryptography`'s dependencies in one place.

File: pallets/runtime/tests/src/transaction_ordering_test.rs [hunk 3]
```diff
@@ -7,21 +7,20 @@ use confidential_asset::{
...
-    mercat::{
-        account::{convert_asset_ids, AccountCreator},
-        asset::AssetIssuer,
-        transaction::{CtxMediator, CtxReceiver, CtxSender},
-        Account, AccountCreatorInitializer, AssetTransactionIssuer, EncryptedAmount,
-        EncryptionKeys, FinalizedTransferTx, InitializedTransferTx, PubAccount, PubAccountTx,
-        SecAccount, TransferTransactionMediator, TransferTransactionReceiver,
-        TransferTransactionSender,
-    },
+    curve25519_dalek::scalar::Scalar,
```
Comment (ParnianAlimi) [inline pallets/runtime/tests/src/transaction_ordering_test.rs:12]: same thing here.

**Review by ParnianAlimi — APPROVED** [review]
Looks good. We should also replace the temp cryptography branch dependencies with a stable one: `develop` or `v2-mainnet`.

**Review by vkomenda — APPROVED** [review]

**Review by fmiguelgarcia — APPROVED** [review]

**Pull request merged**